opensearch-operator/pkg/builders/cluster.go

```diff
@@ -864,6 +865,70 @@ func STSInNodePools(sts appsv1.StatefulSet, nodepools []opsterv1.NodePool) bool
 	return false
 }
 
+func NewSnapshotconfigUpdateJob(
+	instance *opsterv1.OpenSearchCluster,
+	jobName string,
+	namespace string,
+	checksum string,
+	volumes []corev1.Volume,
+	volumeMounts []corev1.VolumeMount,
+) batchv1.Job {
+	httpPort, _ := helpers.VersionCheck(instance)
+	dns := DnsOfService(instance)
+	var snapshotCmd string
+	for _, repository := range instance.Spec.General.Snapshot {
+		var snapshotSettings string
+		for settingsKey, settingsValue := range repository.Settings {
+			snapshotSettings += fmt.Sprintf("\"%s\": \"%s\" , ", settingsKey, settingsValue)
+		}
+		snapshotSettings = strings.TrimRight(snapshotSettings, " ,")
+		snapshotCmd += fmt.Sprintf("curl -s --fail --show-error -k -u \"$(cat /mnt/admin-credentials/username):$(cat /mnt/admin-credentials/password)\" -X PUT https://%s.svc.cluster.local:%v/_snapshot/%s?pretty -H \"Content-Type: application/json\" -d %c{\"type\": \"%s\", \"settings\": {%s}}%c; ", dns, fmt.Sprint(httpPort), repository.Name, '\'', repository.Type, snapshotSettings, '\'')
```

Comment:
great implementation, what happens if client insert a setting that not exist?

Reply:
Hey @idanl21 if the setting does not exists, the error is visible in the job logs and user can take necessary action to re-apply the right config, example as
```
k logs my-first-cluster-snapshotrepoconfig-update-v6ddf -f
{
  "acknowledged" : true
}
{
  "error" : {
    "root_cause" : [
      {
        "type" : "repository_exception",
        "reason" : "[my_s3_repository_4] repository type [s4] does not exist"
      }
    ],
    "type" : "repository_exception",
    "reason" : "[my_s3_repository_4] repository type [s4] does not exist"
  },
  "status" : 500
}
```